### PR TITLE
Fix Preview and Delete Events Batch Job

### PR DESCRIPTION
**Pull request by tn3rb**
While investigating #3893 i discovered that the Preview and Delete Events Batch Job was not advancing properly.

This PR:

- adds new formatting for batch job messages
- adds redirect timers to steps so that users are automatically advanced/returned as needed
- formats and cleans up some of the code

### CODE REVIEW

File: admin_pages/events/form_sections/ConfirmEventDeletionForm.php [hunk 1]
```diff
@@ -69,6 +75,25 @@ public function __construct($event_ids, $options_array = array())
...
+            "
+        <div class='ee-admin-button-row ee-admin-button-row--align-start'>
+            <input type='submit' 
+                   class='button button--primary' 
+                   value='" . esc_attr__('Confirm', 'event_espresso') . "'
+            />
+            <a href='" . esc_url_raw($events_list_url) . ";' class='button button--secondary'>
+                " . esc_html__('Cancel', 'event_espresso') . "
+            </a>
+        </div>"
```
Comment (knazart) [inline admin_pages/events/form_sections/ConfirmEventDeletionForm.php:95]:
```suggestion
            "<div class='ee-admin-button-row ee-admin-button-row--align-start'>
                <input type='submit' 
                       class='button button--primary' 
                       value='" . esc_attr__('Confirm', 'event_espresso') . "'
                />
                <a href='" . esc_url_raw($events_list_url) . ";' class='button button--secondary'>
                    " . esc_html__('Cancel', 'event_espresso') . "
                </a>
            </div>"
```

File: admin_pages/events/templates/event_preview_deletion.template.php [hunk 2]
```diff
@@ -110,13 +110,9 @@
...
-           value="<?php esc_attr_e('Confirm', 'event_espresso'); ?>"
-    />
-    <a href="javascript:window.history.back();" class="button button--secondary">
-        <?php esc_html_e('Cancel', 'event_espresso'); ?>
-    </a>
-</form>
+<?php
+
+echo wp_kses(
+    $form->form_open($form_url) . $form->get_html_and_js() . $form->form_close(),
```
Comment (knazart) [inline admin_pages/events/templates/event_preview_deletion.template.php:116]: 👍

File: core/domain/services/admin/events/data/PreviewDeletion.php [hunk 3]
```diff
@@ -83,73 +85,64 @@ public function __construct(
...
         }
-        $datetime_ids = isset($models_and_ids_to_delete['Datetime']) ? $models_and_ids_to_delete['Datetime'] : array();
-        if (!is_array($datetime_ids)) {
+        $datetime_ids = $models_and_ids_to_delete['Datetime'] ?? [];
+        if (! is_array($datetime_ids)) {
             throw new UnexpectedEntityException($datetime_ids, 'array');
         }
-        $registration_ids = isset($models_and_ids_to_delete['Registration']) ? $models_and_ids_to_delete['Registration'] : array();
-        if (!is_array($registration_ids)) {
+        $registration_ids = $models_and_ids_to_delete['Registration'] ?? [];
```
Comment (knazart) [inline core/domain/services/admin/events/data/PreviewDeletion.php:106]: 👌

**Review by knazart — COMMENTED** [review]
Is there a mistake (one of my review comments), or I'm just missing something ?

File: admin_pages/events/form_sections/ConfirmEventDeletionForm.php [hunk 4]
```diff
@@ -20,21 +23,24 @@
...
     public function __construct($event_ids, $options_array = array())
     {
         if (! isset($options_array['subsections'])) {
             $options_array['subsections'] = [];
         }
-        if (! isset($options_array['subsections']['events'])) {
-            $events_subsection = new \EE_Form_Section_Proper();
-            $options_array['subsections']['events'] = $events_subsection;
+        if (
+            isset($options_array['subsections']['events'])
```
[the remark this answers is not in the document]
Comment (tn3rb) [reply admin_pages/events/form_sections/ConfirmEventDeletionForm.php:38]: ooooo good catch 💯 

this right here... this is it... this is why we do code reviews 👍🏻

File: admin_pages/events/form_sections/ConfirmEventDeletionForm.php (hunk 1 above)
Comment (tn3rb) [reply admin_pages/events/form_sections/ConfirmEventDeletionForm.php:95]: i like to bump the first line down so it doesn't get appended to the previous line in the DOM
makes it easier to read when you look at the actual source code of the page (i'm old and i still do that)

**Pull request merged**